pkg/infra/baseAgent.go

```diff
@@ -262,75 +265,56 @@ func (a *Base) UpdateHPChange(change int) {
-}
+// Updates the utility of a given agent:
+// We take N agents i ∈{1,...,N}that perform the following actions in each iterated round t ∈{1,...,∞}:
+// 1.) Determines the resources it has available, gi ∈[0,1]
+// 2.) Determines its need for resources, qi ∈[0,1]
+// 3.) Makes a provision of resources, pi ∈[0,1]
+// 4.) Makes a demand for resources, di ∈[0,1]
+// 5.) Receives an allocation of resources, ri ∈[0,1]
+// 6.) Makes an appropriation of resources, r′i ∈[0,1]
+
+// The total resources accrued at the end of a round is hence given by:
+// Ri = r′i + (gi −pi)
+// Which provides the utility of an agent, ui, as:
+// ui = αi(qi) + βi(Ri −qi), if Ri ≥qi
+// 		αi(Ri) −γi(qi −Ri), otherwise
+func (a *Base) utility(foodRequested, foodTaken food.FoodType) {
+	// Hyperparameters that are chosen to make utility commensurate for all agents
+	// These initialisations are chosen by Pitt, but may require tuning
+	// Requires that alpha > gamma > beta
+	alpha := 0.2
+	beta := 0.1
+	gamma := 0.18
+
+	// Available resources normalised by total possible available resources
+	g := float64(a.CurrPlatFood()) / float64(a.tower.maxPlatFood)
+
+	// If the agent is on the verge of dying, food is needed
+	var q float64
+	if a.daysAtCritical == a.HealthInfo().MaxDayCritical {
+		q = 1.0
+		// If an agent is not about to die, it effectively needs 0 food
+	} else {
+		q = 0.0
+	}
+	// No resources are recontributed to the common pool (provision = 0)
+	p := 0.0
 
-/*
-HP Gain Factor.
-inputs:
-- gain >= 0
-- loss >= 0
-
-output range:
-- y >= 0.
-
-key values:
-- = 1: agent is sustaining overall
-- < 1: agent is losing hp on average
-- > 1: agent gaining hp on average
-- when loss is 0, default to percentage difference between gain and loss. Acknowledge that HP gain should increase utility
-*/
-// func hpGainFactor(gain, loss, maxhp int) float64 {
-// 	if loss == 0 {
-// 		return 1 + float64(gain)/float64(maxhp)
-// 	}
-// 	return float64(gain) / float64(loss)
-// }
-
-/*
-Net HP Gain Factor.
-This one uses net hp gain as opposed to hp gain/loss ratio.
-*/
-func netHPGainFactor(gain, loss, maxhp int) float64 {
-	netHPGain := gain - loss
-	multiplier := 1 + float64(netHPGain)/float64(maxhp) // reward positive net hp gain, punish otherwise
-	return utilFunctions.RestrictToRange(0, math.Inf(1), multiplier)
-}
+	// Appropriation is the amount of food an agent actually takes, normalised by the max possible
+	r_prime := float64(foodTaken) / float64(a.tower.healthInfo.MaxFoodIntake)
 
-/*
-Food Intake Factor.
-inputs:
-- intake >= 0
-- total >= 0
-range: [0, 1]. == 1, agent has taken all the food it has seen
-
-*/
-func foodIntakeFactor(intake, seen food.FoodType) float64 {
-	// If an agent hasn't seen any food, has 0 utility
-	if seen == 0 {
-		return 0
-	}
-	return float64(intake) / float64(seen)
-}
+	// Note that r isn't included. This is because it is just a metric for assessing Ostrom Institution Theory
+	// r >= r_prime (for conventional institution)
```

Comment:
Irrelevant here. Put it in the report if you want.

Reply:
I think it's important. It's a variable that's included inside of the function description that magically doesn't appear. Having this comment explicitly shows that it's a utility calculation, not a validity calculation

Reply:
I wouldn't put it in the function description then.